### DISCUSSION TITLE
v4.2

### DISCUSSION THREAD

**Discussion by RussKie** [gitextensions/gitextensions#11352] [General]
### Release Notes Highlights
Requires: **.NET 6.0 Desktop Runtime [v6.0.24](https://dotnet.microsoft.com/download/dotnet/thank-you/runtime-desktop-6.0.24-windows-x64-installer)** or later

* Git 2.42 is recommended
* Various performance improvements
* Various UI improvements
* Improvements to Git Commandlog dialog
* Improvements to the Rebase dialog
* Allow "Save as..." multiple files
* Editor can now move line up/down with <kbd>ALT + UP</kbd> and <kbd>ALT + DOWN</kbd>
* OpenSSH to use `SSH_ASKPASS` env variable if a password is requested (requires OpenSSH 8.4 or higher; no effect for older versions of OpenSSH)
* Auto-set GE as editor only in current environment
* Unstaged changes can be reset without affecting staged
* More user friendly handling for "Could not load file or assembly" exceptions
* New GitLab integration plugin
* JIRA plugin now supports personal access tokens
* Vertical tab (<kbd>SHIFT + ENTER</kbd>) is now treated as a line feed
* Add support to build GE for Windows on Arm64 (WoA). This, however, requires manual build.
* New option `{Head}` available in scripts

Full change log: https://github.com/gitextensions/gitextensions/blob/v4.2/GitUI/Resources/ChangeLog.md#version-42-13-nov-2023

![Downloads](https://img.shields.io/github/downloads/gitextensions/gitextensions/v4.2/total.svg)

<hr /><em>This discussion was created from the release <a href='https://github.com/gitextensions/gitextensions/releases/tag/v4.2'>v4.2</a>.</em>

**Comment by hinst** [comment]
No dark modæ? 😔🌒

**Reply by gerhardol** [reply]
This has been discussed several times before, no workaround has been found to override the colors since migration to .NET6.
The best hope is a change in WinForms: https://github.com/dotnet/winforms/issues/7641

**Comment by chirontt** [comment]
A manual build of this v4.2 version for Windows on Arm64 is available in my fork: https://github.com/chirontt/gitextensions/releases/tag/v4.2-win32-aarch64
